tests/func-tests/checkout/widgets-nextjs/tests/commerceWidget.spec.ts

```diff
@@ -0,0 +1,117 @@
+  }
+});
+
+test.describe("widget mounting - connect flow", () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/commerce-connect");
+  });
+
+  test("should render connect widget and handle close", async ({ page }) => {
+    await page.waitForSelector("#widget-root");
+
+    const widgetRoot = page.locator("#widget-root");
+    await expect(widgetRoot).not.toBeEmpty();
+
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByTestId("error-view")).toHaveCount(0);
+
+    const connectWidget = page.getByTestId("connect-wallet");
+    await expect(connectWidget).toBeVisible();
+
+    const closeButton = page.getByTestId("close-button");
+    await expect(closeButton).toBeVisible();
+    await closeButton.click();
+
+    await expect(widgetRoot).toBeEmpty();
+  });
+});
+
+test.describe("widget mounting - bridge flow", () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/commerce-bridge");
+  });
+
+  test("should render bridge widget and handle close", async ({ page }) => {
+    await page.waitForSelector("#widget-root");
+
+    const widgetRoot = page.locator("#widget-root");
+    await expect(widgetRoot).not.toBeEmpty();
+
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByTestId("error-view")).toHaveCount(0);
+
+    const connectWidget = page.getByTestId("bridge-view");
+    await expect(connectWidget).toBeVisible();
+
+    const closeButton = page.getByTestId("close-button");
+    await expect(closeButton).toBeVisible();
+    await closeButton.click();
+
+    await expect(widgetRoot).toBeEmpty();
+  });
+});
+
+/**
+ * Flows behind ConnectLoader require a mocked Provider to get past the ConnectWidget.
+ * Until then, WALLET, SWAP, SALE, ONRAMP won't be testable.
+ */
+test.describe("widget mounting - wallet flow", () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/commerce-wallet");
+  });
+
+  test("should render connect widget", async ({ page }) => {
+    await page.waitForSelector("#widget-root");
+
+    const widgetRoot = page.locator("#widget-root");
+    await expect(widgetRoot).not.toBeEmpty();
+
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByTestId("error-view")).toHaveCount(0);
+
+    const connectWidget = page.getByTestId("connect-wallet");
+    await expect(connectWidget).toBeVisible();
+  });
+});
+
+/**
+ * Add Funds is disabled in Sandbox, we can only validate it doesn't show an error screen.
+ */
+test.describe("widget mounting - add funds flow", () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/commerce-add-funds");
+  });
+
+  test("should render add funds widget", async ({ page }) => {
+    await page.waitForSelector("#widget-root");
+    const widgetRoot = page.locator("#widget-root");
+    await expect(widgetRoot).not.toBeEmpty();
+
+    await page.waitForLoadState('networkidle');
+    await expect(page.getByTestId("error-view")).toHaveCount(0);
```

Comment:
This should be failing, it was locally. `await page.waitForLoadState('networkidle');` doesn't appear to work the same for CI as local.